### PR TITLE
allow for ETS with seasonal=None

### PR DESCRIPTION
**Pull request by john-ramsey**
Fixes: Allow fitting an exponential smoothing model with `seasonal = None`. Ex: Annual Data

### Summary

Currently when trying to fit an ETS model with `seasonal = None` the following error is returned:
`AttributeError: 'NoneType' object has no attribute 'value'`

Example:
```
df = pd.DataFrame(index = pd.date_range(start='1/1/2020', periods=10, freq='A'), data = [i+1 for i in range(10)])
ts = TimeSeries.from_dataframe(df)
mdl = ExponentialSmoothing(seasonal=None)
mdl.fit(ts)
```

### Proposed Fix
Add an if statement which allows for the passing of None into the seasonal component.

### CODE REVIEW

**Comment by dennisbader** [comment]
Hi @john-ramsey and thanks a lot for the PR.

Actually, you can already create a model without seasonality with `ExponentialSmoothing(seasonal=SeasonalityMode.NONE)` or `seasonal=ModelMode.NONE` (see [this answer](https://github.com/unit8co/darts/issues/552#issuecomment-970205490))

The model documention however is a bit misleading especially 

https://github.com/unit8co/darts/blob/4a97140391f171e151c10ac6889eb9a9ce1ba70f/darts/models/forecasting/exponential_smoothing.py#L35

and 

https://github.com/unit8co/darts/blob/4a97140391f171e151c10ac6889eb9a9ce1ba70f/darts/models/forecasting/exponential_smoothing.py#L49-L51

If you want, you could adapt these lines instead for the PR? :)

**Review by dennisbader — CHANGES_REQUESTED** [review]
Hi @john-ramsey and thanks a lot for the PR.

Actually, you can already create a model without seasonality with `ExponentialSmoothing(seasonal=SeasonalityMode.NONE)` or `seasonal=ModelMode.NONE` (see [this answer](https://github.com/unit8co/darts/issues/552#issuecomment-970205490))

The model documention however is a bit misleading especially 

https://github.com/unit8co/darts/blob/4a97140391f171e151c10ac6889eb9a9ce1ba70f/darts/models/forecasting/exponential_smoothing.py#L35

and 

https://github.com/unit8co/darts/blob/4a97140391f171e151c10ac6889eb9a9ce1ba70f/darts/models/forecasting/exponential_smoothing.py#L49-L51

If you want, you could adapt these lines instead for the PR? :)

**Comment by john-ramsey** [comment]
- Clarified a bit by bringing in SeasonalityMode for the ETS seasonal component. Added the None option as well
- Formatted with Black

**Review by dennisbader — CHANGES_REQUESTED** [review]
Thanks a lot for that, looks good!
Just had a few minor suggestions, then it is ready to merge :)

File: darts/models/forecasting/exponential_smoothing.py [hunk 1]
```diff
@@ -20,12 +20,14 @@ def __init__(
         self,
         trend: Optional[ModelMode] = ModelMode.ADDITIVE,
         damped: Optional[bool] = False,
-        seasonal: Optional[ModelMode] = ModelMode.ADDITIVE,
+
```
Comment (dennisbader) [inline darts/models/forecasting/exponential_smoothing.py:23]:
```suggestion
```

File: darts/models/forecasting/exponential_smoothing.py [hunk 2]
```diff
@@ -20,12 +20,13 @@ def __init__(
...
         damped: Optional[bool] = False,
-        seasonal: Optional[ModelMode] = ModelMode.ADDITIVE,
+        seasonal: Optional[ModelMode] = SeasonalityMode.ADDITIVE,
         seasonal_periods: Optional[int] = None,
         random_state: int = 0,
         **fit_kwargs,
     ):
-        """Exponential Smoothing
+
+        """ Exponential Smoothing
```
Comment (dennisbader) [inline darts/models/forecasting/exponential_smoothing.py:29]: for black
```suggestion
        """Exponential Smoothing
```

File: darts/models/forecasting/exponential_smoothing.py [hunk 3]
```diff
@@ -44,13 +45,13 @@ def __init__(
...
         trend
-            Type of trend component. Either ``ModelMode.ADDITIVE`` or ``ModelMode.MULTIPLICATIVE``.
+            Type of trend component. Either ``ModelMode.ADDITIVE``, ``ModelMode.MULTIPLICATIVE``, ``ModelMode.NONE``, or ``None``.
             Defaults to ``ModelMode.ADDITIVE``.
         damped
             Should the trend component be damped. Defaults to False.
         seasonal
-            Type of seasonal component. Either ``ModelMode.ADDITIVE`` or ``ModelMode.MULTIPLICATIVE``.
-            Defaults to ``ModelMode.ADDITIVE``.
+            Type of seasonal component. Either ``SeasonalityMode.ADDITIVE``, ``SeasonalityMode.MULTIPLICATIVE``, ``SeasonalityMode.NONE``, or ``None``.
```
Comment (dennisbader) [inline darts/models/forecasting/exponential_smoothing.py:53]: for black
```suggestion
            Type of seasonal component. Either ``SeasonalityMode.ADDITIVE``, ``SeasonalityMode.MULTIPLICATIVE``, ``SeasonalityMode.NONE``, or ``None``.
```

File: darts/models/forecasting/exponential_smoothing.py [hunk 4]
```diff
@@ -34,8 +35,8 @@ def __init__(
 
         `model_mode` must be a ``ModelMode`` Enum member. You can access the Enum with ``from darts import ModelMode``.
 
-        ``ExponentialSmoothing(trend=None, seasonal=None)`` corresponds to a single exponential smoothing.
-        ``ExponentialSmoothing(trend=ModelMode.ADDITIVE, seasonal=None)`` corresponds to a Holt's exponential smoothing.
+        ``ExponentialSmoothing(trend=ModelMode.NONE, seasonal=SeasonalityMode.NONE)`` corresponds to a single exponential smoothing.
+        ``ExponentialSmoothing(trend=ModelMode.ADDITIVE, seasonal=SeasonalityMode.NONE)`` corresponds to a Holt's exponential smoothing.
```
Comment (dennisbader) [inline darts/models/forecasting/exponential_smoothing.py:39]:
```suggestion
        ``ExponentialSmoothing(trend=ModelMode.NONE, seasonal=SeasonalityMode.NONE)`` corresponds to a single
        exponential smoothing.
        ``ExponentialSmoothing(trend=ModelMode.ADDITIVE, seasonal=SeasonalityMode.NONE)`` corresponds to a Holt's
        exponential smoothing.
```

File: darts/models/forecasting/exponential_smoothing.py [hunk 5]
```diff
@@ -44,13 +45,13 @@ def __init__(
         Parameters
         ----------
         trend
-            Type of trend component. Either ``ModelMode.ADDITIVE`` or ``ModelMode.MULTIPLICATIVE``.
+            Type of trend component. Either ``ModelMode.ADDITIVE``, ``ModelMode.MULTIPLICATIVE``, ``ModelMode.NONE``, or ``None``.
             Defaults to ``ModelMode.ADDITIVE``.
```
Comment (dennisbader) [inline darts/models/forecasting/exponential_smoothing.py:49]:
```suggestion
            Type of trend component. Either ``ModelMode.ADDITIVE``, ``ModelMode.MULTIPLICATIVE``, ``ModelMode.NONE``,
            or ``None``. Defaults to ``ModelMode.ADDITIVE``.
```

File: darts/models/forecasting/exponential_smoothing.py [hunk 6]
```diff
@@ -44,13 +45,13 @@ def __init__(
...
         damped
             Should the trend component be damped. Defaults to False.
         seasonal
-            Type of seasonal component. Either ``ModelMode.ADDITIVE`` or ``ModelMode.MULTIPLICATIVE``.
-            Defaults to ``ModelMode.ADDITIVE``.
+            Type of seasonal component. Either ``SeasonalityMode.ADDITIVE``, ``SeasonalityMode.MULTIPLICATIVE``, ``SeasonalityMode.NONE``, or ``None``.
+            Defaults to ``SeasonalityMode.ADDITIVE``.
```
Comment (dennisbader) [inline darts/models/forecasting/exponential_smoothing.py:54]:
```suggestion
            Type of seasonal component. Either ``SeasonalityMode.ADDITIVE``, ``SeasonalityMode.MULTIPLICATIVE``,
            ``SeasonalityMode.NONE``, or ``None``. Defaults to ``SeasonalityMode.ADDITIVE``.
```

**Comment by john-ramsey** [comment]
I have black auto-formatting in VS code and it caught none of this? 😬

Thanks

**Comment by dennisbader** [comment]
Most of the issues like too long lines were detected by flask8 that we just integrated into CI.
However unit tests seem to behave strangely currently.  This seems to be an issue on our side

**Review by dennisbader — APPROVED** [review]
All tests passed now. Thanks again for contributing!

**Pull request merged**